### ISSUE TITLE
Question about model initialization

### ISSUE THREAD

**Issue by w00zie** [locuslab/ect#10]
Hi, first of all thank you for your work and open-sourcing the code!

I have a question regarding Table 1 in your manuscript: when you report FID values for ECM on CIFAR-10 do you initialize ECM with EDM / perform a separate pre-training with r/t = 0 or it is a model trained completely from scratch ? If I understand correctly, ECT stands for pretraining with the diffusion objective while ECM stands for fine-tuning with the full objective ? If it is the case, can you share some tips on how to train ECT models (e.g. to reproduce Table 5 in the appendix) ?

Congratulations for your work !

**Comment by w00zie** [comment]
Inspecting Fig. 2, you state that you have a ~32x training budget speedup wrt to iCT.

According to iCT authors, they train their CIFAR-10 models with 1024 batch size * 400k iterations ~ 410M training images.
Your model, on the other hand, is trained with 12.8M images that is, approximately, 1/32 of iCT.

Since iCT models are trained from scratch and your ECM model is (I believe) initialized with a pre-trained EDM, wouldn't it be more correct to state that your training budget is 200M (EDM) + 12.8M (ECM), thus approximately 1/2 of iCT ? (and still outperforming them by a large margin)

Thank you again !

**Comment by Gsunshine** [comment]
Hi @w00zie ,

Thanks for your interest in ECT! 

Fig.2 compares with iCT-deep, which is a 2x deeper model. The overall cost is around 1/4. Fig. 2 gives a clear breakdown of each part regarding pretraining & tuning. I've updated the banner figure on the GitHub repo to Fig. 2 used in the paper, which skips 1/32. 

I think more seriously the efficiency offers opportunities to reflect on how to take advantage of findings in tuning to improve the entire pipeline (pretraining + tuning).

---

> Initialize ECM with EDM / perform a separate pre-training with r/t = 0 / Is it a model trained completely from scratch? 

I didn't see differences here. r/t = 0 falls back to diffusion pretraining. EDM is a standard diffusion model w/ x-prediction parameterization (data prediction). I'd say r/t = 0 eq to EDM or other score/diffusion/flow pretraining schemes, where differences are up to weighting functions or noise distributions.

ECT (for methods) & ECM (resulting models) have no differences in the context. Pretraining + Tuning is also a scheme of training from scratch, using a single model throughout. 

Have fun with the consistency-diffusion family!

Thanks,
Zhengyang

**Comment by Gsunshine** [comment]
If you want to reduce the overall cost further while maintaining comparable performance to iCT, consider cutting the pretraining budget, too. 

Basically, you could do

1. Improved performance & lower cost.
2. Similar performance & further lower cost.

It is moving along the Pareto frontier of performance & compute.

**Comment by w00zie** [comment]
Thank you @Gsunshine for your answer!

I think I understand your point and I agree with it but it is still not clear to me how to perform the full training from scratch pipeline, i.e. pre-training and fine-tuning.

It is unclear to me how to perform the pre-training phase with your code: while there is the code for finetuning (by loading a pre-trained EDM model) it is less clear (to me) how to pre-train with the r = 0 objective.

In your code https://github.com/locuslab/ect/blob/cb3536063a51b9fc1d8e64dccb1cc2f448ab0fa6/training/ct_training_loop.py#L382 you update `new_stage` where `cur_tick` is updated at each training iteration and `double_ticks` is set to, e.g., 250. This means that the (pre-training) r = 0 stage lasts a few training steps only, is that correct?

Do you have some guidelines / reference values for setting `double_ticks` so that it can lead to a convergent pre-training phase ? A training script to replicate Table 5 would also help a lot!

I hope my answer is sufficiently clear but I can elaborate more if needed.

Thank you again!

**Comment by w00zie** [comment]
Hey @Gsunshine I was wondering if you had the time to check this issue, sorry to bother !

Thank you !

**Comment by 1999kevin** [comment]
Same question! Any guidelines if we do not have a pre-training model on a special dataset?  Do we need to first run the dataset with EDM to obtain a pre-training model before using eCM?
